src/main/java/meteordevelopment/meteorclient/systems/modules/player/SpeedMine.java

```diff
@@ -7,20 +7,38 @@
 
 import meteordevelopment.meteorclient.events.world.TickEvent;
 import meteordevelopment.meteorclient.mixin.StatusEffectInstanceAccessor;
+import meteordevelopment.meteorclient.settings.BlockListSetting;
 import meteordevelopment.meteorclient.settings.DoubleSetting;
 import meteordevelopment.meteorclient.settings.EnumSetting;
 import meteordevelopment.meteorclient.settings.Setting;
 import meteordevelopment.meteorclient.settings.SettingGroup;
 import meteordevelopment.meteorclient.systems.modules.Categories;
 import meteordevelopment.meteorclient.systems.modules.Module;
 import meteordevelopment.orbit.EventHandler;
+import net.minecraft.block.Block;
 import net.minecraft.entity.effect.StatusEffectInstance;
 
+import java.util.List;
+
 import static net.minecraft.entity.effect.StatusEffects.HASTE;
 
 public class SpeedMine extends Module {
     private final SettingGroup sgGeneral = settings.getDefaultGroup();
 
+    private final Setting<List<Block>> blocks = sgGeneral.add(new BlockListSetting.Builder()
+            .name("blocks")
+            .description("Selected blocks.")
+            .filter(block -> block.getHardness() > 0)
```

Comment:
Why this filter?

Reply:
Some anticheats will only allow speedmine to work on combat blocks like obsidian, so you'd have to turn it on and off when you want to mine normal blocks like stone

Reply:
So you don't get unnecessary blocks like end portal, bedrock, nether portal, etc. in blocks list